src/relalg/cross_bench.cc

```diff
@@ -7,21 +7,21 @@
 
 namespace fluent {
 
-void ManualSimpleBench(benchmark::State& state) {
+
+void ManualTupleBench(benchmark::State& state) {
   while (state.KeepRunning()) {
     state.PauseTiming();
     std::vector<int> xs(state.range_x());
     std::vector<int> ys(state.range_x());
     state.ResumeTiming();
     for (int x : xs) {
       for (int y : ys) {
-        benchmark::DoNotOptimize(x);
-        benchmark::DoNotOptimize(y);
+        std::make_tuple(x, y);
```

Comment:
You should call `benchmark::DoNotOptimize` on this or else the whole loop might get optimized away.